lib/tesla/adapter/gun.ex

```diff
@@ -553,5 +553,10 @@ if Code.ensure_loaded?(:gun) do
           ip
       end
     end
+
+    # it is reasonable to make a post request with an empty body,
+    # however gun does not natively support it
+    defp format_request_body(%{method: :post} = env), do: env.body || "{}"
```

Comment:
@alex-strizhakov What would be the correct solution here? Using `{}` doesn't seem 100% _right_

Reply:
I don't get why `gun` doesn't support `nil` or `""` values for body requests.
```elixir
client = Tesla.client([Tesla.Middleware.JSON], Tesla.Adapter.Gun)
Tesla.post(client, "http://httpbin.org/post", "", headers: [{"content-type", "application/json"}])
```
```{:ok,
 %Tesla.Env{
   __client__: %Tesla.Client{
     adapter: {Tesla.Adapter.Gun, :call, [[]]},
     fun: nil,
     post: [],
     pre: [{Tesla.Middleware.JSON, :call, [[]]}]
   },
   __module__: Tesla,
   body: %{
     "args" => %{},
     "data" => "",
     "files" => %{},
     "form" => %{},
     "headers" => %{
       "Content-Length" => "0",
       "Content-Type" => "application/json",
       "Host" => "httpbin.org",
       "X-Amzn-Trace-Id" => "Root=1-6293805a-6d06c446640ad7a776aae5a7"
     },
     "json" => nil,
     "origin" => "176.232.61.138",
     "url" => "http://httpbin.org/post"
   },
   headers: [
     {"date", "Sun, 29 May 2022 14:16:58 GMT"},
     {"content-type", "application/json"},
     {"content-length", "332"},
     {"connection", "keep-alive"},
     {"server", "gunicorn/19.9.0"},
     {"access-control-allow-origin", "*"},
     {"access-control-allow-credentials", "true"}
   ],
   method: :post,
   opts: [],
   query: [],
   status: 200,
   url: "http://httpbin.org/post"
 }}
```
Same goes for `nil`.
@gaiabeatrice Can you please elaborate on what's wrong with `gun`?